### ISSUE TITLE
Only updates bookmark titles

### ISSUE THREAD

**Issue by SLK350** [dgutov/bmreplace#40]
When I click on toolbar button for bm replacer it only updates the title of the bookmark not the new url. Been doing this for a few days now using Nightly build 56 x64 windows.

**Comment by dgutov** [comment]
Thanks for the report, I'm still on beta.

Anything interesting in the error console when this happens?

**Comment by SLK350** [comment]
Only update of bookmark title seem to work (after clicking ok to change title). When I try to update both url & title nothing happens, but when I try to update only title it works.

This comes up in error console:

NS_ERROR_ILLEGAL_VALUE: Invalid value for tags

**Comment by dgutov** [comment]
> NS_ERROR_ILLEGAL_VALUE: Invalid value for tags

Thanks. Does it have a reference to a file and a line?

Alternatively, could you try applying this patch locally? `.xpi` files are basically zip files; you'll have to edit inside it:

```diff
diff --git a/content/bookmarks.js b/content/bookmarks.js
index 05fa01c..002fce5 100644
--- a/content/bookmarks.js
+++ b/content/bookmarks.js
@@ -167,7 +167,7 @@ let bm = {
    */
   replaceBookmark: function(id, url, title, description, usePrivateBrowsing) {
     let oldUri = bms.getBookmarkURI(id),
-        tags = ts.getTagsForURI(oldUri, {}),
+        tags = ts.getTagsForURI(oldUri),
         uri = ios.newURI(url, null, null);
     ts.tagURI(uri, tags);
     ts.untagURI(oldUri, tags);
```

**Comment by dgutov** [comment]
> Will this extension become an webextension later?

Planning to, yes. I'll probably stay on Firefox 56 for a while, though, until some Tab Groups alternative comes out.

**Comment by SLK350** [comment]
Could you send me the new xpi. karl-johan [at] tanno.se

**Comment by dgutov** [comment]
Sent.

**Comment by SLK350** [comment]
Still not working.

**Comment by dgutov** [comment]
Does the problem occur with every bookmark, or are there particular pages that trigger this?

**Comment by SLK350** [comment]
All pages.

**Comment by dgutov** [comment]
I've just downloaded Firefox 56.0a1 and launched it with a test profile. The bookmarks seem to be moving okay (url included), even those that have tags.

Can you reproduce the problem step by step in an empty profile?

**Comment by dgutov** [comment]
This is the version I've tried: 56.0a1 (2017-07-13) (64-bit).

**Comment by SLK350** [comment]
Tried with a new profile and still wont work. What I tried was:

1. Go to "https://forums.mydigitallife.net/threads/discussion-windows-10-insider-preview-build-16241-pc.74707/" Page 1 and make a bookmark.
2. Go to "https://forums.mydigitallife.net/threads/discussion-windows-10-insider-preview-build-16241-pc.74707/page-2" page 2 and click on "Replace Bookmark" toolbar button. Click ok to update bookmark.
3. When clicking on bookmark afterwards it still goes to page 1 and title still reads "Page 1".

So neither url or name is updated.

If I try to just update the name of the bookmark if it's been changed it will work.
